custom-metrics-stackdriver-adapter/pkg/adapter/translator/query_builder.go

```diff
@@ -509,9 +512,20 @@ func (t *Translator) ListMetricDescriptors(fallbackForContainerMetrics bool) *st
 		Filter(filter)
 }
 
+type cachedMetricInfo struct {
```

Comment:
replace it with LRUExpireCache https://pkg.go.dev/k8s.io/apimachinery/pkg/util/cache

Reply:
Thanks, updated. I put some initial values here that i think would work. Let me know if we want to make this configurable.

Reply:
> Let me know if we want to make this configurable.

Yes, it will be better to make it configurable. And this feature can be turned off by default.

If we find more users are complaining this issue, we can make it turn on by default later.

Reply:
MetricKindCacheSize:
- When it's 0, it doesn't use the cache
- When it's not 0, it uses the cache